LochNessBuilder/BuilderRegistry.cs

```diff
@@ -6,40 +6,93 @@
+
+            var builderTypes = new List<Type>();
+            foreach (var assembly in possiblyRelevantAssemblies)
+            {
+                try
+                {
+                    var builderDecoratedTypes =
+                        assembly
+                            .GetExportedTypes()
+                            .Where(t => t.GetCustomAttribute<BuilderAttribute>() != null)
+                            .ToList();
+                    builderTypes.AddRange(builderDecoratedTypes);
+                }
+                catch
+                {
+                    // Suppress any errors, since this functionality isn't critical, and almost all errors here will
+                    // be about assemblies that are in some way innaccessible, and hence weren't interesting assemblies.
+                }
+            }
+
+            return builderTypes;
+        }
+
+        private static void RegisterMostAppropriateGetter(Type builderType)
         {
-            Type builderType;
+            var publicStaticGetProperties =
+                builderType.GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.GetProperty);
+            var propsReturningABuilder = publicStaticGetProperties
+                .Where(prop =>
+                    prop.PropertyType.IsGenericType &&
+                    prop.PropertyType.GetGenericTypeDefinition() == typeof(Builder<>))
+                .ToList();
 
-            // ReSharper disable once ConvertIfStatementToConditionalTernaryExpression
-            if (Builders.ContainsKey(typeof(TInstance)))
+            switch (propsReturningABuilder.Count)
             {
-                builderType = Builders[typeof(TInstance)];
+                case 0:
+                    throw new NotImplementedException(
+                        $"The type {builderType} is marked as a builder, but has no public static getter returning a Builder<>");
+                case 1:
+                    RegisterPropertyGetterAsBuilderFactory(propsReturningABuilder.Single());
+                    break;
+                default:
+                    // There are more than one candidate Prop. Pick one, use it, ignore the others.
+                    var propsCalledNew = propsReturningABuilder.Where(prop => prop.Name == "New").ToList();
+                    var propToUse = propsCalledNew.FirstOrDefault() ?? propsReturningABuilder.First();
+                    RegisterPropertyGetterAsBuilderFactory(propToUse);
+                    break;
             }
-            else
+        }
+
+        private static void RegisterPropertyGetterAsBuilderFactory(PropertyInfo propReturningABuilder)
+        {
+            var builderGetMethod = propReturningABuilder.GetGetMethod();
+            var typeBuilt = builderGetMethod.ReturnType.GetGenericArguments()[0];
+            BuilderFactoryMethods.Add(typeBuilt, builderGetMethod);
```

Comment:
Discussed & Agreed.
Resolved with Dictionary<Type, List<MethodInfo>

Reply:
now throws:
`                throw new NotSupportedException($"There are multiple BuilderFactories defined for type '{typeof(TInstance).FullName}'. Please use .WithBuilder() to specify which one should be used, rather than `WithBuilt()` since it is unable to infer the correct one.");`